File: youtubeapi/src/main/java/com/liskovsoft/youtubeapi/feedback/FeedbackApi.java
package com.liskovsoft.youtubeapi.feedback;

import com.liskovsoft.googlecommon.common.converters.jsonpath.WithJsonPath;
import com.liskovsoft.youtubeapi.feedback.models.FeedbackResponse;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;

/**
 * For signed users!
 */
@WithJsonPath
public interface FeedbackApi {
    @Headers("Content-Type: application/json")
    @POST("https://www.youtube.com/youtubei/v1/feedback")
    Call<FeedbackResponse> setNotInterested(@Body String feedbackQuery);
}

File: youtubeapi/src/main/java/com/liskovsoft/youtubeapi/actions/FeedbackApi.kt
package com.liskovsoft.youtubeapi.actions

import com.liskovsoft.youtubeapi.actions.models.HideVideoTokenResult
import com.liskovsoft.youtubeapi.common.helpers.DefaultHeaders
import retrofit2.Call
import retrofit2.http.Headers
import retrofit2.http.POST

interface FeedbackApi {
    @Headers("Content-Type: application/json", "User-Agent: " + DefaultHeaders.USER_AGENT_CHROME)
    @POST("https://www.youtube.com/youtubei/v1/browse")
    fun getHideVideoToken(videoId: String): Call<HideVideoTokenResult>
}